editors/code/src/toolchain.ts

```diff
@@ -180,13 +181,11 @@ function lookupInPath(exec: string): boolean {
     return candidates.some(isFileAtPath);
 }
 
-async function isFileAtPath(path: string): Promise<boolean> {
-    return isFileAtUri(vscode.Uri.file(path));
-}
-
-async function isFileAtUri(uri: vscode.Uri): Promise<boolean> {
+function isFileAtPath(suspectPath: string): boolean {
+    // It is not mentionned in docs, but `statSync()` throws an error when
+    // the path doesn't exist
     try {
-        return ((await vscode.workspace.fs.stat(uri)).type & vscode.FileType.File) !== 0;
+        return fs.statSync(suspectPath).isFile();
```

Comment:
I think this shouldn't switch from `vscode.workspace.fs` to `fs`. I think something like `Promise.all(candidates.map(isFileAtPath)).some((isFile) => isFile)` instead of `candidates.some(isFileAtPath)` would work.

Reply:
`Promise.all` still need an async execution context thus `getPathForExecutable` must be made async and a lot of public functions depending on `getPathForExecutable` must be made async.

Reply:
I have changed all the calling functions to async.